rpc_util.go

```diff
@@ -610,9 +613,11 @@ func (p *parser) recvMsg(maxReceiveMessageSize int) (pf payloadFormat, msg []byt
 	if int(length) > maxReceiveMessageSize {
 		return 0, nil, status.Errorf(codes.ResourceExhausted, "grpc: received message larger than max (%d vs. %d)", length, maxReceiveMessageSize)
 	}
-	// TODO(bradfitz,zhaoq): garbage. reuse buffer after proto decoding instead
-	// of making it for each message:
-	msg = make([]byte, int(length))
+	if p.sharedRecvBufferPool != nil {
```

Comment:
What if we `s/shared//` in this whole PR and make a default implementation that just does `make` on every call to `Get` (w/`Put` a nop) and avoids the need for these conditionals?

Reply:
Done with 15f820e51f3622eb0a79185fd985d117f4f1c044